c7n/resources/kms.py

```diff
@@ -440,6 +440,66 @@ def format_resource(self, r):
         return envelope
 
 
+@Key.filter_registry.register('last-rotation')
+class LastRotation(ValueFilter):
+    """Queries KMS keys by the last time they were rotated.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: kms-not-rotated-in-last-30
+                resource: kms-key
+                filters:
+                  - type: last-rotation
+                    key: RotationDate
+                    value: 30
+                    value_type: age
+                    op: gte
+
+    """
+
+    schema = type_schema('last-rotation', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('kms:ListKeyRotations',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('kms')
+        results = []
+
+        for r in resources:
+            if 'c7n:LastRotation' in r:
+                # We've already got a last-rotation on the resource. Skip the
+                # API call.
+                continue
```

Comment:
Missed this at first, if the annotation already exists we don't need to re-fetch rotation data but we _do_ still need to evaluate the filter and return resources that match. 

Can follow up with more detail or suggestions, just wanted to explain why I immediately dismissed my last approval.